VerilogCodingStyle.md

```diff
@@ -2291,6 +2292,109 @@ always_comb begin
 end
 ```
 
+### Functions and Tasks
+
+The following section applies to synthesizable RTL only.
+
+***In synthesizable RTL use of functions is allowed, provided they are declared
+automatic. Tasks should not be used.***
+
+Functions must be declared outside of a class in either a package or inside a
+module. A package is appropriate where the function relates to other definitions
+in the package and could be useful to multiple modules (even if its currently
+only used by one). A module is appropriate where the function specifically
+relates to the internals of that module.
+
+Functions should aim to conceptually represent a reusable block of combinational
+logic.
+
+Storage types must be explicitly declared for all arguments and the function
```

Comment:
Perhaps limit the args to only use 4-state data type `logic`?

Reply:
+1

Reply:
I think it's useful to be able to pass enums and structs etc so I'll expand that to 4-state data type `logic` and datatypes derived from `logic`.

Also I think it's useful to have functions that work on integer parameters (a classic example is log2 though we have an in-built function for that) but to get this PR landed I'll forgot about that for now and add an issue.